compiler/src/iree/compiler/ConstEval/Runtime.cpp

```diff
@@ -149,13 +141,19 @@ void CompiledBinary::deinitialize() {
 
 FunctionCall::FunctionCall(CompiledBinary &binary, iree_host_size_t argCapacity,
                            iree_host_size_t resultCapacity)
-    : binary(binary) {
-  IREE_CHECK_OK(iree_vm_list_create(iree_vm_make_undefined_type_def(),
-                                    argCapacity, iree_allocator_system(),
-                                    &inputs));
-  IREE_CHECK_OK(iree_vm_list_create(iree_vm_make_undefined_type_def(),
-                                    resultCapacity, iree_allocator_system(),
-                                    &outputs));
+    : binary(binary), argCapacity(argCapacity), resultCapacity(resultCapacity) {
+}
+
+LogicalResult FunctionCall::initialize() {
+  if (!iree_status_is_ok(iree_vm_list_create(iree_vm_make_undefined_type_def(),
```

Comment:
you can't nest status creation with isok as you leak the status
you need to use a macro that does the right thing or a function that converts and consumes for you - you'd iree_status_ignore or ideally log and then free before returning your LogicalResult. You can use InFlightDiagnostic and such to append the status and return the failure.

I think handleRuntimeError does most of that